mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -193,6 +199,42 @@ uploadSingle file a = do
   x <- liftIO $ LBS.readFile file
   void . send $ fencode' putObject a (toBody x) & poServerSideEncryption .~ pure sse
 
+-- | Upload a file from a URL to a given S3 Address
+uploadFromURL :: URL -> Address -> AWS UploadResult
+uploadFromURL = uploadFromURLWithMode Fail
+
+-- | Upload a file from a URL to a given S3 Address
+--   fail if the destination already exists and the WriteMode is Fail
+uploadFromURLWithMode :: WriteMode -> URL -> Address -> AWS UploadResult
+uploadFromURLWithMode m url a = do
+  e <- ask
+  eitherT (pure . UploadError) (const $ pure UploadOk) $ do
+  when (m == Fail) . whenM (lift $ exists a) . left $ UploadDestinationExists a
+  EitherT (Right <$> do
+    let manager = e ^. envManager
+    request  <- liftIO $ parseUrl (exportURL url)
+    response <- http request manager
+    let body =  responseBody response
+    let chunk = 100 * 1024 * 1024
+    body $$+- uploadSink a chunk)
+
+-- | Create an upload sink for a given S3 Address
+uploadSink :: Address -> Int -> Sink BS.ByteString AWS ()
```

Comment:
I think you could possible achieve the same outcome here without the `chunksOfAtLeast` by using [ChunkedBody](https://hackage.haskell.org/package/amazonka-core-1.3.2/docs/Network-AWS-Data-Body.html#t:ChunkedBody) as the [RqBody](https://hackage.haskell.org/package/amazonka-core-1.3.2/docs/Network-AWS-Data-Body.html#t:RqBody) for `putObject` where you can specify the chunk size etc, might be worth looking into